src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -0,0 +1,143 @@
+package exoticatechnologies.hullmods.exotics
+
+import com.fs.starfarer.api.combat.ShipVariantAPI
+import com.fs.starfarer.api.fleet.FleetMemberAPI
+import exoticatechnologies.modifications.exotics.impl.HullmodExotic
+import exoticatechnologies.util.AnonymousLogger
+import exoticatechnologies.util.datastructures.Optional
+
+object HullmodExoticHandler {
+    private val lookupMap: HashMap<HullmodExoticKey, HullmodExoticInstallData> = hashMapOf()
+
+    /**
+     * Method for checking whether the [HullmodExotic] should be installed onto a [ShipVariantAPI] by checking
+     * if we hadn't already installed it or not
+     *
+     * @param hullmodExotic the HullmodExotic in question
+     * @param parentFleetMember the [FleetMemberAPI] of the root module, or the whole ship's if it's a single-module ship
+     * @param variant the variant to check
+     * @param variantList list of module variants belonging to the parent module (optional)
+     */
+    fun shouldInstallHullmodExoticToVariant(hullmodExotic: HullmodExotic, parentFleetMember: FleetMemberAPI, variant: ShipVariantAPI, variantList: Optional<List<ShipVariantAPI>>): Boolean {
```

Comment:
improve javadoc to better explain *when* the `Optional` must be present and will be used.

Also some class javadoc to explain it's purpose.

Reply:
still not done

Reply:
done. resolving.